### PR TITLE
Replacing missing CPASTATE in historical audits

### PR DESCRIPTION
**Pull request by phildominguez-gsa**
Closes https://github.com/GSA-TTS/FAC/issues/3711

In this PR:
- Some historic audits are missing values for the `CPASTATE` field in the eleccpas table, which (correctly) fails our validations.
- This fix replaces the missing values with GSA_MIGRATION and updates the secondary auditors schema validations to allow this value
- Disallowing GSA_MIGRATION in uploaded workbooks by adding `secondary_auditor_address_state` to `check_gsa_migration_keyword.py`
- Creating `append_secondary_auditor_changes` for making inspection records
- Unit tests

Testing:
- Find an audit in the eleccpas table with a blank `CPASTATE`, or clear one that already exists
  - There are no modifiable entries when using local data, but you can create a row with these values, which are mostly borrowed from the one failing in preview: `7627,"243753201823516","2016","58817","","",FAC OVERRIDE,FAC OVERRIDE,FAC OVERRIDE,"","99999",FAC OVERRIDE,FAC OVERRIDE,"2133184987","",FAC OVERRIDE,"999999999"`
- Run `historic_data_migrator` on that audit and it should migrate without issue
- The entry in the migrationinspectionrecord table should show it replaced "" with "GSA_MIGRATION"
- `address_state` in the dissemination_secondary_auditor table should be "GSA_MIGRATION"

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/census_historical_migration/test_secondary_auditors.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from django.conf import settings
```
Comment (gsa-suk) [inline backend/census_historical_migration/test_secondary_auditors.py:1]: A different file name of 'test_secondary_auditor_xforms.py' might match the rest of the test files.

Comment (phildominguez-gsa) [reply backend/census_historical_migration/test_secondary_auditors.py]: Done

**Review by gsa-suk — APPROVED** [review]
lgtm.

File: backend/census_historical_migration/workbooklib/secondary_auditors.py [hunk 2]
```diff
@@ -59,6 +64,30 @@
...
 
 
+def xform_address_state(secondary_auditors):
+    """Default missing address_state to GSA_MIGRATION"""
+    change_records = []
+    is_empty_address_state_found = False
+
+    for secondary_auditor in secondary_auditors:
+        address_state = string_to_string(secondary_auditor.CPASTATE)
+        if not address_state:
```
Comment (sambodeme) [inline backend/census_historical_migration/workbooklib/secondary_auditors.py:74]: We want to call `track_transformations` regardless of `if not address_state` condition. However, we want save the record (i.e., `InspectionRecord.append_secondary_auditor_changes(change_records)`) only if is_empty_address_state_found` is true.  @phildominguez-gsa

**Pull request merged**